Branch-SDK/src/main/java/io/branch/referral/AttributionReportingManager.kt

```diff
@@ -0,0 +1,138 @@
+                            val manager = MeasurementManager.get(context)
+
+                            val branchBaseURL = PrefHelper.getInstance(context).apiBaseUrl
+                            val requestJson = request.toJSON().getJSONObject("REQ_POST")
+                            var triggerUri = createURIFromJSON(branchBaseURL, requestJson)
+
+                            if (triggerUri.toString().length > 10000) {
+                                val trimmedJson = trimJsonForUri(requestJson)
+                                val trimmedUri = createURIFromJSON(branchBaseURL, trimmedJson)
+
+                                BranchLogger.d("Trigger URI is too long. Removed content items from request JSON. New URI: $trimmedUri")
+                                triggerUri = trimmedUri
+                            }
+
+                            //Add Sandbox version to the trigger URI
+                            val sandboxVersion = SdkExtensions.getExtensionVersion(SdkExtensions.AD_SERVICES).toString()
+                            triggerUri = triggerUri.buildUpon().appendQueryParameter(Defines.Jsonkey.Privacy_Sandbox_Version.key, sandboxVersion).build()
+
+                            manager.registerTrigger(triggerUri, executor, object : OutcomeReceiver<Any?, Exception> {
+                                    override fun onResult(result: Any?) {
+                                        BranchLogger.v("Trigger registered successfully with URI: $triggerUri")
+                                        executor.shutdown()
+                                    }
+
+                                    override fun onError(e: Exception) {
+                                        BranchLogger.w("Error while registering trigger with URI $triggerUri: ${e.message}")
+                                        executor.shutdown()
+                                    }
+                                }
+                            )
+                        } else {
+                            BranchLogger.v("Measurement API is not enabled. Did not register trigger.")
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                BranchLogger.w("Error while registering source: ${e.message}")
+            } finally {
+                scope.cancel()
+            }
+        }
+    }
+
+    private fun createURIFromJSON(baseURL: String, json: JSONObject): Uri {
+        val builder = Uri.parse(baseURL).buildUpon()
+        val params = mutableMapOf<String, String>()
+        parseJson("", json, params)
+
+        params.forEach { (key, value) ->
+            builder.appendQueryParameter(URLEncoder.encode(key, "UTF-8"), URLEncoder.encode(value, "UTF-8"))
+        }
+
+        return builder.build()
+    }
+
+    // Recursive function to flatten the JSON object into a map of strings
+    private fun parseJson(prefix: String, json: JSONObject, params: MutableMap<String, String>) {
```

Comment:
I would rename to `flattenJsonForRegisterTriggerUri` or something more descriptive since `parseJson` is rather generic.